velodyne_pointcloud/src/lib/rawdata.cc

```diff
@@ -458,14 +494,21 @@ namespace velodyne_rawdata
             intensity = (intensity > max_intensity) ? max_intensity : intensity;
 
             if (pointInRange(distance)) {
+              // Set point time as beginning of scan and then apply timing offset:
+              ros::Time pt_time = pkt.stamp;
+              if (!timing_offsets_.empty()) {
+                  pt_time = pt_time + ros::Duration(timing_offsets_[block][firing_seq]);
```

Comment:
I'm wondering about the performance of constructor of `ros::Duration`. It's in the innermost place of this loop (that processes ~ 1 million points per second).

ros::Time is a subclass of some base class also, idk if there is a vtable lookup to access `sec` and `nsec` probably not (or it doesn't matter due to branch prediction). We should run intel vtunes for this node just to make sure there is nothing that surprises us in terms of performance.

Reply:
Why not make your `timing_offsets_` be `vector<vector<Duration>>`?

Reply:
fixed.